src/util/assert_or_die.h

```diff
@@ -0,0 +1,16 @@
+/*******************************************************************\
+
+Module: Assert or die
+
+Author: Diffblue Ltd
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_ASSERT_OR_DIE_H
+#define CPROVER_UTIL_ASSERT_OR_DIE_H
+
+#include "std_code.h"
+
+codet create_assert_or_die(const exprt &condition, const source_locationt &loc);
```

Comment:
1) I think `expr_util` or `std_code` are perfectly good places to put this, no need for a new file.
2) Be specific about the return type. There is no need to hide the fact that it returns a `code_blockt`.
3) I'm not sure I like the name. How about `aborting_assert`?